src/data-interfaces/HoloFuelDnaInterface.js

```diff
@@ -195,57 +230,110 @@ const HoloFuelDnaInterface = {
+      if (transactionArray.length === 0) {
+        throw new Error(`no pending transaction with id ${transactionId} found.`)
+      } else {
+        return transactionArray[0]
+      }
+    },
+    /* NOTE: This is to allow handling of the other side of the transaction that was declined.  */
+    getPendingDeclined: async (transactionId) => {
+      const declinedResult = await createZomeCall('transactions/list_pending_declined')({ origins: transactionId })
+      const transactionArray = declinedResult.map(presentDeclinedTransaction)
+
       if (transactionArray.length === 0) {
         throw new Error(`no pending transaction with id ${transactionId} found.`)
       } else {
         return transactionArray[0]
       }
     },
-    // decline pending proposed transaction (NB: proposed by another agent).
+    /* NOTE: decline ACTIONABLE TRANSACTION (NB: pending transaction proposed by another agent) >> ONLY for on asynchronous transactions. */
     decline: async (transactionId) => {
       const transaction = await HoloFuelDnaInterface.transactions.getPending(transactionId)
-      await createZomeCall('transactions/decline')({ origin: transactionId })
+      const declinedProof = await createZomeCall('transactions/decline_pending')({ origins: transactionId })
+      if (!declinedProof) throw new Error('Decline Error.', declinedProof)
       return {
         ...transaction,
         id: transactionId,
-        status: STATUS.rejected
+        status: STATUS.declined
       }
     },
-    // cancel pending authored transaction.
+    /* NOTE: cancel WAITING TRANSACTION that current agent authored. */
     cancel: async (transactionId) => {
-      const transaction = await HoloFuelDnaInterface.transactions.getPending(transactionId)
-      await createZomeCall('transactions/cancel')({ origin: transactionId })
+      const authoredRequests = await HoloFuelDnaInterface.transactions.allNonActionableByState(transactionId, ['incoming/requested', 'outgoing/approved'])
+      const transaction = authoredRequests.find(authoredRequest => authoredRequest.id === transactionId)
+      const reason = annulTransactionReason
+      const canceledProof = await createZomeCall('transactions/cancel_transactions')({ origins: transactionId, reason })
+      if (!canceledProof) throw new Error('Decline Error.', canceledProof)
+
       return {
         ...transaction,
         id: transactionId,
-        status: STATUS.cancelled
+        status: STATUS.canceled
       }
     }
   },
+  /* NOTE: recover funds from DECLINED PENDING TRANSACTION (ie: Counterparty declined offer) - intended for REFUNDS  */
+  recoverFunds: async (transactionId) => {
+    const reason = annulTransactionReason
+    const transaction = await HoloFuelDnaInterface.transactions.getPendingDeclined(transactionId)
+    const canceledProof = await createZomeCall('transactions/cancel')({ entry: transaction, reason })
```

Comment:
there are two endpoints in the DNA called `transactions/cancel` and `transactions/cancel_transactions` and they do different things?

Reply:
Yes, they do. The prior is for a proactive cancel, allowing the initiating agent to cancel an offer/request prior to the counterparty taking action.   The later is in response to a decline action by the counterparty - basically for restoring any the money from cancelled offer back to the initiating agent's balance.